test/module/irohad/consensus/yac/yac_gate_test.cpp

```diff
@@ -199,7 +201,8 @@ TEST_F(YacGateTest, CacheReleased) {
   outcome_notifier.get_subscriber().on_next(expected_commit);
   round.reject_round++;
 
-  gate->vote({boost::none, round, ledger_state});
+  block_notifier.get_subscriber().on_next(
```

Comment:
please look at this code of `YacGateImpl` constructor:
https://github.com/hyperledger/iroha/blob/da82b960430dcbbb60726cb298f1e404fb5af72d/irohad/consensus/yac/impl/yac_gate_impl.cpp#L70-L73

if the line updating `current_ledger_state_` could be moved to `YacGateImpl::vote(const BlockCreatorEvent &)`, which seems to improve consistency without breaking any existing logic, you could also avoid this change.

Reply:
Looks great, thanks!